File: agent/llm_node.py
import os

from pydantic import SecretStr
from langchain_openai import ChatOpenAI

from agent.state import AgentState
from agent.tools import ALL_TOOLS

llm = ChatOpenAI(
    model=os.environ.get("MODEL", "openai/gpt-5-nano"),
    base_url="https://openrouter.ai/api/v1",
    api_key=SecretStr(os.environ["OPENROUTER_API_KEY"]),
    temperature=0.2,
).bind_tools(ALL_TOOLS)


def llm_node(state: AgentState) -> dict:
    response = llm.invoke(state["messages"])
    return {"messages": [response]}

File: src/agent/llmNode.ts
import { ChatOpenAI } from "@langchain/openai";
import { MessagesAnnotation } from "@langchain/langgraph";
import { ALL_TOOLS } from "./tools.js";

const model = new ChatOpenAI({
  model: process.env.MODEL ?? "openai/gpt-5-nano",
  configuration: {
    baseURL: "https://openrouter.ai/api/v1",
  },
  openAIApiKey: process.env.OPENROUTER_API_KEY!,
  temperature: 0.2,
}).bindTools(ALL_TOOLS);

export async function llmNode(state: typeof MessagesAnnotation.State) {
  const response = await model.invoke(state.messages);
  return { messages: [response] };
}
